ereolen_support/ereolen_support.form.inc

```diff
@@ -0,0 +1,167 @@
+  );
+
+  $form['ereolen_support_product'] = array(
+    '#type' => 'select',
+    '#title' => t('Product'),
+    '#options' => _ereolen_support_create_select_list('ereolen_support_product'),
+    '#description' => variable_get('ereolen_support_product_description'),
+    '#required' => variable_get('ereolen_support_product_description_required')
+  );
+
+  $form['ereolen_support_problem'] = array(
+    '#type' => 'select',
+    '#title' => t('Problem'),
+    '#options' => _ereolen_support_create_select_list('ereolen_support_problem'),
+    '#description' => variable_get('ereolen_support_problem_description'),
+    '#required' => variable_get('ereolen_support_problem_required')
+  );
+
+  $form['ereolen_support_description'] = array(
+    '#type' => 'textarea',
+    '#title' => t('Description'),
+    '#description' => variable_get('ereolen_support_description_description'),
+    '#required' => variable_get('ereolen_support_description_required')
+  );
+
+  $form['ereolen_support_date'] = array(
+    '#type' => 'date_popup',
+    '#title' => t('Dato'),
+    '#date_format' => 'd/m/Y',
+    '#date_label_position' => 'none',
+    '#description' => variable_get('ereolen_support_date_description'),
+    '#required' => variable_get('ereolen_support_date_required')
+  );
+
+  $form['ereolen_support_book_title'] = array(
+    '#type' => 'textfield',
+    '#title' => t('What book does the problem concern?'),
+    '#description' => variable_get('ereolen_support_book_title_description'),
+    '#required' => variable_get('ereolen_support_book_title_required')
+  );
+
+  $form['ereolen_support_device'] = array(
+    '#type' => 'select',
+    '#title' => t('What device did you use?'),
+    '#options' => _ereolen_support_create_select_list('ereolen_support_device'),
+    '#description' => variable_get('ereolen_support_device_description'),
+    '#required' => variable_get('ereolen_support_device_required')
+  );
+
+  $form['ereolen_support_model'] = array(
+    '#type' => 'textfield',
+    '#title' => t('On what device model did you experience the problem?'),
+    '#description' => variable_get('ereolen_support_model_description'),
+    '#required' => variable_get('ereolen_support_model_required')
+  );
+
+  $form['ereolen_support_operating_system'] = array(
+    '#type' => 'textfield',
+    '#title' => t('On what operating system did you experience the problem?'),
+    '#description' => variable_get('ereolen_support_operating_system_description'),
+    '#required' => variable_get('ereolen_support_model_operating_system')
+  );
+
+  $form['submit_button'] = array(
+    '#type' => 'submit',
+    '#value' => t('Send'),
+  );
+
+  return $form;
+}
+
+/**
+ * Add a submit handler to the support form.
+ *
+ * @param $form
+ *   The form to submit.
+ * @param $form_state
+ *   The state of the form.
+ *
+ * @throws \Exception
+ */
+function ereolen_support_form_submit($form, &$form_state) {
+  global $user;
```

Comment:
Error handling back to the user if the jira stuff fails?

Reply:
After re-reading the all the code... I see that as a side-effect you will get user feedback, but as a side effect. The jira functions e.i. `_ereolen_support_create_jira_user` should throw an exception and the user interaction (display_message) should be handled here. If we look into "Single-responsibility principle" to simplify the code and make maintains easier from SOLID.